packages/next/src/build/create-compiler-aliases.ts

```diff
@@ -233,6 +193,46 @@ export function createServerOnlyClientOnlyAliases(
       }
 }
 
+export function createNextApiEsmAliases() {
+  const mapping = {
+    head: 'next/dist/api/head',
+    image: 'next/dist/api/image',
+    constants: 'next/dist/api/constants',
+    router: 'next/dist/api/router',
+    dynamic: 'next/dist/api/dynamic',
+    script: 'next/dist/api/script',
+    link: 'next/dist/api/link',
+    navigation: 'next/dist/api/navigation',
+    headers: 'next/dist/api/headers',
+    og: 'next/dist/api/og',
+    // pages api
+    document: 'next/dist/api/document',
+    app: 'next/dist/api/app',
```

Comment:
Probably a silly question, but can you clarify why it's ok for this to be `app` and `document` if previously`dist/esm/_app` and `dist/esm/_document`?

Reply:
previously we were locating the page `dist/esm/_app|_document`, now with the new `api/` folder, they're re-exporting everything from `dist/(esm/)?pages/_app|_document`. So we were mapping file path to another file path

In this way, it will be easier to maintain as we only need to care about the api path `next/app` instead of the file path `next/dist/pages/_app`. The esm mapping takes effect with above folder mapping